tests/dragonfly/conftest.py

```diff
@@ -85,6 +85,20 @@ def df_server(df_factory: DflyInstanceFactory) -> DflyInstance:
     assert clients_left == 1
 
 
+@pytest.fixture(scope="session")
+def df_cluster_server(df_factory: DflyInstanceFactory) -> DflyInstance:
```

Comment:
@dranikpg  it seems this function is redundant. am I missing something?

Reply:
Yes, we probably should not make any distinction between "default" instances and cluster mode instances (as long this is just a single-instance reply-only implementation)

@lsvmello
If you look at `dfly_args`, you'll see that it annotates the `df_factory` fixture. Each time those parameter change, the `df_server` is re-created. So only one of df_server and df_cluster_server can be running at a time (because they will have the same port). And if only one of two exists at a time, we can re-use the first 😃

Reply:
@dranikpg I duplicated cause I'm having an issue with the assertion on the end of df_server fixture and didn't want to affect other tests:
```python
instance.stop()
assert clients_left == 1
```
Don't know if it because I'm not using the fixture properly (first time ever on python), but my tests end up with 3 clients_left and I have 3 tests functions on the same class.

Reply:
In your new fixture `cluster_client`, you can replace
```py
return client
```

with 

```py
yield client
client.disconnect_connection_pools()
```

This will first return the client and then run the cleanup after tests have finished. 

We currently try not to re-use connections between tests to minimize interference and prevent any forgotten pub-sub listeners, etc.